adagrams/game.py

```diff
@@ -1,11 +1,128 @@
+from audioop import add
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+
+
+
 def draw_letters():
-    pass
+    letters = []
+
+    while len(letters) < 10:
+        random_letter = random.choice(list(LETTER_POOL.keys()))
+        
+        if letters.count(random_letter) < LETTER_POOL[random_letter]:
+            letters.append(random_letter)
+
+    return letters
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    available_letters = letter_bank[:]
+    upper_word = word.upper()
+
+    for letter in upper_word:
+        if letter not in available_letters:
+            return False
+        else:
+            available_letters.remove(letter)
+
+    return True
+
+
 
 def score_word(word):
-    pass
+    score = 0
+    upper_word = word.upper()
+
+    for letter in upper_word:
+        if letter in SCORE_CHART:
+            score += SCORE_CHART[letter]
+
+    if len(word) >= 7:
+        score += 8
```

Comment:
Looks like the test didn't pick this up but you should be checking if the length of the word is between 6 and 11.